### ISSUE TITLE
Suggestion: Add Proxy Settings

### ISSUE THREAD

**Issue by m2afs** [RipMeApp/ripme#285]
* Ripme version:1.7.3
* Java version: 8
* Operating system: win 10

Well as the title says its a suggestion not an issue, so right now in order to use proxy we have to change windows setting, how about adding proxy detection to the ripper, even if its manual its still effective enough.
you can detect proxy from Internet Explorer. 👍 

tnx anyway

**Comment by cyian-1756** [comment]
I agree this would be a good setting to have

**Comment by metaprime** [comment]
I think we've previously said that this should be an OS-level setting. Not sure if we can choose to use a proxy server from an app (going around the OS setting for proxy).

Also not sure how to set up a proxy server to test this functionality.

How do you detect a proxy in Internet Explorer?

You only want ripme to detect a proxy (os-level) is in use? Why? What's your use case here that an OS-level proxy doesn't solve?

**Comment by m2afs** [comment]
@metaprime 
yeah i only want the rip me to detect my current proxy settings or have some boxes for ports

similar to IDM 
in their setting Downloads>Options>Proxy/Socks

We can fill whatever we want the program to use or it can detect from Internet Option itself.

**Comment by m2afs** [comment]
also found this https://stackoverflow.com/questions/4933677/detecting-windows-ie-proxy-setting-using-java

don't know if its helping or not

**Comment by metaprime** [comment]
Oh, does ripme currently ignore the OS proxy settings?

**Comment by m2afs** [comment]
yeah it does,
check this free proxy https://www.psiphon3.com/
it connects and changes the settings in Internet Options but rip me ignores them or maybe uses different ports or diffrent protocols i dont exactly know, look at it urself :)

**Comment by fabianschneider78** [comment]
i think there are two options:
- setting a proxy using System.setProperty (https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html)
- replacing every connection with
`SocketAddress address = new InetSocketAddress("urlOfAGreatProxy.com", 1080);
Proxy proxy = new Proxy(Proxy.Type.SOCKS, address);
URL url = new URL(urlStr);
URLConnection conn = url.openConnection(proxy);`

I have got no experience if the first approach is really working, but the second would require a lot of refractoring (some connections are established directly, some use Http.java and some JSoup)

One could test this easily by creating a local proxy in java.

**Comment by fabianschneider78** [comment]
I just found out that jsoup has had proxy support for quite some time now:
https://github.com/jhy/jsoup/commit/4384e1692572ad8ea554eabd1f110a13191812c3
https://github.com/jhy/jsoup/commit/23982c287bf6efc453a22bcbbf8c955786a74b92

That would make the implementation of the feature a lot easier.
But first the jsoup dependency needs to be updated.
